src/tower.ts

```diff
@@ -42,6 +43,52 @@ export class PisaTower {
         // start watching it if it passed inspection
         await this.watcher.addAppointment(appointment);
 
-        return appointment;
+        const signature = await this.appointmentSigner.signAppointment(appointment);
+        return { appointment, signature };
+    }
+}
+
+/**
+ * This class is responsible for signing Ethereum appointments.
+ */
+export abstract class EthereumAppointmentSigner {
+    /**
+     * Signs `appointment`. Returns a promise that resolves to the signature of the appointment.
+     *
+     * @param appointment
+     */
+    public abstract signAppointment(appointment: EthereumAppointment): Promise<string>;
+}
+
+/**
+ * This EthereumAppointmentSigner signs appointments using a hot wallet.
+ */
+export class HotEthereumAppointmentSigner extends EthereumAppointmentSigner {
+    constructor(private readonly signer: ethers.Signer) {
+        super();
+    }
+
+    /**
+     * Signs `appointment`. Returns a promise that resolves to the signature of the appointment.
+     *
+     * @param appointment
+     */
+    public signAppointment(appointment: EthereumAppointment): Promise<string> {
+        // TODO: this signature format does not match Pisa's contract
```

Comment:
Lets remove this TODO from the code, and add it as a ticket.

Reply:
Done.